### DISCUSSION TITLE
ZAP Scan on Argo Workflow UI reports CSP: script-src unsafe-inline & CSP: style-src unsafe-inline

### DISCUSSION THREAD

**Discussion by ramanNarasimhan77** [argoproj/argo-workflows#12671] [Q&A]
Hello,
We have scanned Argo workflow UI using [OWASP ZAP](https://www.zaproxy.org/)

CSP reported is `default-src 'self' 'unsafe-inline'; img-src 'self'`

ZAP tool reports this as a security issue.  
Could you please provide some context on why this is set in the Argo Workflow UI? 

References:
https://content-security-policy.com/unsafe-inline/ 
[feat(ui): Enable CSP, HSTS, X-Frame-Options. Fixes #2760, #1376, #276… · argoproj/argo-workflows@9f86a4e (github.com)](https://github.com/argoproj/argo-workflows/commit/9f86a4e941ecca4399267f7780fbb2e7ddcd2199#diff-62729366a4240a4ed4809a45271985740e65297229d8538e4ba4971891556dbbR31)

**Comment by terrytangyuan** [comment]
cc @agilgur5 do you know?

**Reply by agilgur5** [reply]
I only looked at briefly so far at #2971 etc (note my reaction to the opening comment and labeling -- I'd have assigned myself if I could have, but this is a discussion, not an issue).
It's on my todo list to check more extensively. Otherwise I think we should refrain from any public statements until we have certainty given possible security ramifications.

**Reply by ramanNarasimhan77** [reply]
@agilgur5 Should I convert this into an issue? I thought of first asking here to know if it's done intentionally (incase some functionalities in UI do not work if this is not set)

**Reply by agilgur5** [reply]
Technically it should be a private, draft security disclosure 😅 Then we'd investigate to see what the ramifications would be and if any patches are necessary therein or if it's a false positive etc. That would likely effectively end up on my plate anyway though (as a UI and Security SME).
